libcalico-go/lib/backend/model/block_affinity.go

```diff
@@ -39,9 +39,22 @@ const (
 	StatePendingDeletion BlockAffinityState = "pendingDeletion"
 )
 
+type AffinityConfig struct {
+	AffinityType AffinityType
+	Host         string
+}
+
+type AffinityType string
+
+const (
+	AffinityTypeHost    AffinityType = "host"
+	AffinityTypeVirtual AffinityType = "virtual"
+)
+
 type BlockAffinityKey struct {
-	CIDR net.IPNet `json:"-" validate:"required,name"`
-	Host string    `json:"-"`
+	CIDR        net.IPNet      `json:"-" validate:"required,name"`
+	Host        string         `json:"-"`
+	AffinityCfg AffinityConfig `json:"-"`
```

Comment:
I think this is a bit of a blurring of responsibilities here.

the `type BlockAffinityKey` is already effectively the model representation of an `AffinityConfig`.

I would have expected:

- AffinityConfig stays in the `ipam` package as the IPAM code's way of representing / passing around affinity configuration in its API and internally.
- `BlockAffinityKey` simply gets a new field - `Type` - to represent the new (previously implied to be "host") data that identifies a unique block.